text/0115-circuit-breaker.md

```diff
@@ -0,0 +1,110 @@
+# Title
+
+- RFC PR: https://github.com/tikv/rfcs/pull/115
+- Tracking Issue: https://github.com/tikv/pd/issues/8678
+
+## Summary
+
+Implement a [circuit breaker](https://learn.microsoft.com/en-us/azure/architecture/patterns/circuit-breaker) pattern to protect the PD leader from overloading due to retry storms or similar feedback loops.
+
+## Motivation
+
+In large TiDB clusters with hundreds of tidb and tikv nodes, the PD leader can become overwhelmed during certain failure conditions, leading to a "retry storm" or other feedback loop scenarios. 
+Once triggered, PD transitions into a metastable state and cannot recover autonomously, leaving the cluster in a degraded or unavailable state.
+This feature is especially critical for large clusters where a high number of nodes can continuously hammer a single PD instance during failures, leading to cascading effects that worsen recovery times and overall cluster health.
+
+## Detailed design
+
+### Configuration
+
+The key part of the circuit breaker is its configuration. 
+In order to limit number of system variables, we will use a single system variables per circuit breaker type to define error rate threshold. 
+
+#### System variables
+
+`tidb_cb_pd_metadata_error_rate_threshold_pct`:
+- Scope: GLOBAL
+- Persists to cluster: Yes
+- Applies to hint SET_VAR: No
+- Type: Integer
+- Default value: 0
+- Range: [0, 100]
+- This variable is used to set percent of errors to trip the circuit breaker.`0` means circuit breaker is disabled.
+
+#### Hardcoded configs
+
+* `error_rate_window`
+* Defines how long to track errors before evaluating error_rate_threshold.
+* Default: 10
+* Unit: seconds
+---
+* `min_qps_to_open`
```

Comment:
Do we need a config for it?

Reply:
If "had only 1 request within error_rate_window", you do not have to enable the cb?

Reply:
lgtm